ownCloud/Client/Actions/Actions+Extensions/DuplicateAction.swift

```diff
@@ -39,34 +39,32 @@ class DuplicateAction : Action {
 			return
 		}
 
-		guard let rootItem = item.parentItem(from: core) else {
-			completed(with: NSError(ocError: .itemNotFound))
-			return
-		}
+		context.items.forEach({ (item) in
+			if let rootItem = item.parentItem(from: core) {
 
-		var name: String = "\(itemName) copy"
+				var name: String = "\(itemName) copy"
 
-		if item.type != .collection {
-			if let itemFileExtension = item.fileExtension, let baseName = item.baseName {
-				var fileExtension = itemFileExtension
+				if let itemFileExtension = item.fileExtension, let baseName = item.baseName {
+					var fileExtension = itemFileExtension
 
-				if fileExtension != "" {
-					fileExtension = ".\(fileExtension)"
-				}
+					if fileExtension != "" {
+						fileExtension = ".\(fileExtension)"
+					}
 
-				name = "\(baseName) copy\(fileExtension)"
-			}
-		}
+					name = "\(baseName) copy\(fileExtension)"
```

Comment:
There is no fallback, if `"\(baseName) copy\(fileExtension)"` also exists. This is a general problem by our app, not only here. Maybe we need a function for checking, if a file name exists. If it exists, we need a new name `copy copy` or `copy 2` and recheck this name again, until it succeeds.

Reply:
@hosy I actually created separate issue for that, since I think it makes sense to keep stuff separated and not to try to solve issues which are out of scope for now.

Reply:
Did not see that you already created an issue for that. Thanks!